connection.go

```diff
@@ -0,0 +1,132 @@
+package papergres
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+// SSLMode defines all possible SSL options
+type SSLMode string
+
+const (
+	// SSLDisable - No SSL
+	SSLDisable SSLMode = "disable"
+	// SSLRequire - Always SSL, no verification
+	SSLRequire SSLMode = "require"
+	// SSLVerifyCA - Always SSL, verifies that certificate was signed by trusted CA
+	SSLVerifyCA SSLMode = "verify-ca"
+	// SSLVerifyFull - Always SSL, verifies that certificate was signed by trusted CA
+	// and server host name matches the one in the certificate
+	SSLVerifyFull SSLMode = "verify-full"
+)
+
+// Connection holds all database connection configuration.
+type Connection struct {
+	Database    string
+	User        string
+	Password    string
+	Host        string
+	Port        string
+	AppName     string
+	Timeout     int
+	SSLMode     SSLMode
+	SSLCert     string
+	SSLKey      string
+	SSLRootCert string
+}
+
+// NewConnection creates and returns the Connection object to the postgres server
+func NewConnection(databaseURL string, appName string) Connection {
+	u, err := url.Parse(databaseURL)
+	if err != nil {
+		panic(err)
+	}
+
+	host, port, _ := net.SplitHostPort(u.Host)
+	p, _ := u.User.Password()
+	q, _ := url.ParseQuery(u.RawQuery)
+	path := u.Path
+	if strings.Index(path, "/") == 0 {
+		path = path[1:]
+	}
+
+	// default
+	sslMode := SSLDisable
+	if len(q["sslmode"]) > 0 && q["sslmode"][0] != "" {
```

Comment:
expressions like this can be placed in a function that does one thing and returns the results. Doing this makes it easier to read, maintain and test.

Reply:
I completely agree with that! Thanks moved it to a function call in new commit.